bittide/src/Bittide/ProcessingElement.hs

```diff
@@ -41,44 +42,53 @@ processingElement ::
   ( HiddenClockResetEnable dom
   , KnownNat nBusses, 2 <= nBusses, CLog 2 nBusses <= 30) =>
   PeConfig nBusses ->
-  Vec (nBusses-2) (Signal dom (WishboneS2M (Bytes 4))) ->
-  Vec (nBusses-2) (Signal dom (WishboneM2S (32 - CLog 2 nBusses) 4 (Bytes 4)))
-processingElement config bussesIn = case config of
-  PeConfig memMapConfig initI initD ->
-    go memMapConfig initI initD
+  Circuit () (Vec (nBusses-2) (Wishbone dom 'Standard (MappedBus 32 nBusses) (Bytes 4)))
+processingElement (PeConfig memMapConfig initI initD) = circuit $ do
+  (iBus, dBus) <- rvCircuit (pure low) (pure low) (pure low)
+  ([iMemBus, dMemBus], extBusses) <-
+    (splitAtCircuit d2 <| singleMasterInterconnect memMapConfig) -< dBus
+  wbStorage initI -< iBus
+  wbStorageDPC initD -< (iMemBus, dMemBus)
+  idC -< extBusses
+
+splitAtCircuit ::
```

Comment:
```suggestion
-- | Conceptually the same as 'splitAt', but for 'Circuit's
splitAtC ::
```

Reply:
I think we suffix all the other circuit specific helpers with `C` in `clash-protocols`.